cmd/osquery-perf/agent.go

```diff
@@ -2135,12 +2135,17 @@ func (a *agent) runLiveYaraQuery(query string) (results []map[string]string, sta
 	}
 	request.Header.Add("Content-type", "application/json")
 
-	// Make the request.  For load testing purposes we don't actually care about the response.
+	// Make the request.
 	response, err := http.DefaultClient.Do(request)
 	if err != nil {
 		ss := fleet.OsqueryStatus(1)
 		return []map[string]string{}, &ss, ptr.String(fmt.Sprintf("yara request failed to run: %v", err)), nil
 	}
+	// For load testing purposes we don't actually care about the response, but check that we at least got one.
+	if _, err := io.Copy(io.Discard, response.Body); err != nil {
+		ss := fleet.OsqueryStatus(1)
+		return []map[string]string{}, &ss, ptr.String(fmt.Sprintf("error reading response from yara API: %v", err)), nil
+	}
 	defer response.Body.Close()
```

Comment:
`defer response.Body.Close()` should be before the `io.Copy`.

Reply:
good catch, fixed